pepper-apis/src/main/java/org/broadinstitute/ddp/event/publish/pubsub/EventPubSubPublisher.java

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.ddp.event.publish.pubsub;
+
+import static java.lang.String.format;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_TASK_TYPE;
+
+import java.util.Map;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutureCallback;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.rpc.ApiException;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import org.broadinstitute.ddp.constants.ConfigFile;
+import org.broadinstitute.ddp.event.publish.EventPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implementation of {@link EventPublisher} providing publishing
+ * an event of a specified type to Google PubSub topic.<br>
+ * The parameters for PubSub publishing are taken from config:
+ * `pubsub.eventActionTopic` - name of PubSub topic where to publish event.
+ */
+public class EventPubSubPublisher implements EventPublisher {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventPubSubPublisher.class);
+
+    public static final String ATTR_STUDY_GUID = "studyGuid";
+    public static final String ATTR_PARTICIPANT_GUID = "participantGuid";
+
+    @Override
+    public void publishEvent(
+            String eventType,
+            String payload,
+            String studyGuid,
+            String participantGuid) {
+
+        var publisherData = PubSubPublisherInitializer.getOrCreatePubSubPublisherData(
+                ConfigFile.PUBSUB_EVENT_ACTION_TOPIC);
```

Comment:
What happens if we want to send a message to DSM outside of an event action? I recommend renaming this to something else.

Reply:
@yufengwng you mean to rename ConfigFile parameter?

Reply:
Not just ConfigFile parameter, but in general.

Reply:
Fixed. Refactored